src/object/custom_particle_system.cpp

```diff
@@ -1241,4 +1246,1024 @@ CustomParticleSystem::ease_value(float* value, float target, float time, easing
+
+// =============================================================================
+//   Delay
+// -----------------------------------------------------------------------------
+
+float
+CustomParticleSystem::get_delay() const
+{
+  return m_delay;
+}
+
+void
+CustomParticleSystem::set_delay(float delay)
+{
+  m_delay = delay;
+}
+
+void
+CustomParticleSystem::fade_delay(float delay, float time)
+{
+  ease_value(&m_delay, delay, time, getEasingByName(EasingMode::EaseNone));
+}
+
+void
+CustomParticleSystem::ease_delay(float delay, float time, const std::string& easing_)
+{
+  ease_value(&m_delay, delay, time, getEasingByName(EasingMode_from_string(easing_)));
+}
+
+// =============================================================================
+//   Lifetime
+// -----------------------------------------------------------------------------
+
+float
+CustomParticleSystem::get_lifetime() const
+{
+  return m_particle_lifetime;
+}
+
+void
+CustomParticleSystem::set_lifetime(float lifetime)
+{
+  m_particle_lifetime = lifetime;
+}
+
+void
+CustomParticleSystem::fade_lifetime(float lifetime, float time)
+{
+  ease_value(&m_particle_lifetime, lifetime, time, getEasingByName(EasingMode::EaseNone));
+}
+
+void
+CustomParticleSystem::ease_lifetime(float lifetime, float time, const std::string& easing_)
+{
+  ease_value(&m_particle_lifetime, lifetime, time, getEasingByName(EasingMode_from_string(easing_)));
+}
+
+// =============================================================================
+//   Lifetime variation
+// -----------------------------------------------------------------------------
+
+float
+CustomParticleSystem::get_lifetime_variation() const
+{
+  return m_particle_lifetime_variation;
+}
+
+void
+CustomParticleSystem::set_lifetime_variation(float lifetime_variation)
+{
+  m_particle_lifetime_variation = lifetime_variation;
+}
+
+void
+CustomParticleSystem::fade_lifetime_variation(float lifetime_variation, float time)
+{
+  ease_value(&m_particle_lifetime_variation, lifetime_variation, time, getEasingByName(EasingMode::EaseNone));
+}
+
+void
+CustomParticleSystem::ease_lifetime_variation(float lifetime_variation, float time, const std::string& easing_)
+{
+  ease_value(&m_particle_lifetime_variation, lifetime_variation, time, getEasingByName(EasingMode_from_string(easing_)));
+}
+
+// =============================================================================
+//   Birth time
+// -----------------------------------------------------------------------------
+
+float
+CustomParticleSystem::get_birth_time() const
+{
+  return m_particle_birth_time;
+}
+
+void
+CustomParticleSystem::set_birth_time(float birth_time)
+{
+  m_particle_birth_time = birth_time;
+}
+
+void
+CustomParticleSystem::fade_birth_time(float birth_time, float time)
+{
+  ease_value(&m_particle_birth_time, birth_time, time, getEasingByName(EasingMode::EaseNone));
+}
+
+void
+CustomParticleSystem::ease_birth_time(float birth_time, float time, const std::string& easing_)
+{
```

Comment:
Can the parameter be renamed to `easing` or does that conflict with a class member?

Reply:
It conflicts with a typedef